### PR TITLE
fix bug that start default worker failed when args contains whitespace

### PR DESCRIPTION
**Pull request by tomsunelite**
## Why are these changes needed?
Fix the issue that java worker cannot start correctly when the arguments contains space.

## Related issue number

#28331

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [x] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by jovany-wang** [comment]
Could you add a test for it? like adding some whitespace in an option.

```java
Ray.actor(A::new).setJvmOptions(List.of("-DA=B1 B2")).remote();
```

**Comment by jovany-wang** [comment]
Hi, is there any update?

**Comment by tomsunelite** [comment]
hello, who can review this pr ?

**Comment by jovany-wang** [comment]
@tomsunelite I take this. Is this ready for review?

File: java/test/src/main/java/io/ray/test/SetupJavaWorkerTest.java
Comment (jovany-wang) [inline java/test/src/main/java/io/ray/test/SetupJavaWorkerTest.java:43]: For this class, I suggest that:
1. use `extends BaseTest` instead of `Ray.init()` and `Ray.shutdown()` manually to make case could be exited gracefully everytime.
2. Add a `@BeforeClass` annotation to the setup method.

So, I believe this case could be refined to:
```java
@Test
public class SetupJavaWorkerTest extends BaseTest {

  @BeforeClass
  public void setup() {
    String oldCodeSearchPathStr = System.getProperty("java.class.path");
    // mock a fake code search path contains space
    String newCodeSearchPathStr = oldCodeSearchPathStr + ":fakepathStart fakePathEnd";
    System.setProperty("java.class.path", newCodeSearchPathStr);
  }


  public static class Counter {

    private int value;

    public Counter(int initValue) {
      this.value = initValue;
    }

    public int getValue() {
      return value;
    }
  }

  /**
   * raylet will call setup_worker.py with arguments to start java workers after Ray 2.0.0, and
   * setup_worker.py can not start java workers correctly when argumens contian space. This test
   * case is used to test whether the bug is fixed.
   */
  public void testCreateActorFail()
      throws InterruptedException, NoSuchFieldException, IllegalAccessException {
    ActorHandle<ActorTest.Counter> actor = Ray.actor(ActorTest.Counter::new, 1).remote();
    // throw RayTimeoutException exception if actor is not started correctly by raylet
    Integer value = actor.task(ActorTest.Counter::getValue).remote().get(3 * 1000);
    Assert.assertEquals(Integer.valueOf(1), value);
  }
}
```

File: java/test/src/main/java/io/ray/test/WorkerJvmOptionsTest.java
Comment (jovany-wang) [inline java/test/src/main/java/io/ray/test/WorkerJvmOptionsTest.java:28]: How about adding a new case `testJvmOptionsWithWhitespace()` instead of change this case?
Otherwise, this case will be changed by other guys in the future, with ignoring the whitespace.

**Review by jovany-wang — COMMENTED** [review]
Thanks, it looks good overall. 2 minor comments should be addressed before merging.

File: java/test/src/main/java/io/ray/test/WorkerJvmOptionsTest.java
Comment (tomsunelite) [reply java/test/src/main/java/io/ray/test/WorkerJvmOptionsTest.java:28]: sure, but seems that this case is used to test if the argument contains whitespace according to the comments:
// The whitespaces in following argument are intentionally added to test
// that raylet can correctly handle dynamic options with whitespaces.

**Review by jovany-wang — APPROVED** [review]
Looks good to me.

**Comment by tomsunelite** [comment]
blocked by ci ?

**Comment by jovany-wang** [comment]
CI failure is unrelated. merging it.
Thanks for your contribution on this!

**Pull request merged**
